src/netex/read.rs

```diff
@@ -0,0 +1,709 @@
+    c.dates.insert(day);
+}
+
+fn read_stop_assignements(
+    _collections: &mut Collections,
+    context: &mut NetexContext,
+    stop_assignments: &Element,
+) {
+    for node in stop_assignments.children() {
+        // assuming all children are PassengerStopAssignment
+        context.routepoint_mapping.insert(
+            node.get_child("ScheduledStopPointRef", &context.namespace)
+                .unwrap()
+                .attr("ref")
+                .unwrap()
+                .to_string(),
+            node.get_child("QuayRef", &context.namespace)
+                .unwrap()
+                .attr("ref")
+                .unwrap()
+                .to_string(),
+        );
+    }
+}
+
+fn read_time_table_frame(
+    collections: &mut Collections,
+    context: &mut NetexContext,
+    time_table_frame: &Element,
+) -> Result<()> {
+    let vj_node = time_table_frame
+        .get_child("vehicleJourneys", &context.namespace)
+        .unwrap();
+    read_vehicle_journeys(collections, context, &vj_node);
+    Ok(())
+}
+
+fn read_organisations(
+    collections: &mut Collections,
+    context: &mut NetexContext,
+    organisations: &Element,
+) -> Result<()> {
+    let companies: Vec<_> = organisations
+        .children()
+        .map(|node| objects::Company {
+            id: node.attr("id").unwrap().to_string(),
+            name: node
+                .get_child("Name", &context.namespace)
+                .unwrap()
+                .text()
+                .to_string(),
+            address: None,
+            url: None,
+            mail: None,
+            phone: None,
+        })
+        .collect();
+    if companies.len() > 0 {
+        context.first_operator_id = companies[0].id.to_string();
+        let mut companies: Vec<objects::Company> = companies
+            .into_iter()
+            .filter(|c| collections.companies.get_idx(&c.id).is_none())
+            .collect();
+        collections.companies.append(&mut companies)?;
+    } else {
+        context.first_operator_id = "default_company".to_string();
+        if collections
+            .companies
+            .get_idx(&context.first_operator_id)
+            .is_none()
+        {
+            collections
+                .companies
+                .push(objects::Company {
+                    id: "default_company".to_string(),
+                    name: "Default Company".to_string(),
+                    address: None,
+                    url: None,
+                    mail: None,
+                    phone: None,
+                })
+                .unwrap();
+        };
+    }
+    Ok(())
+}
+
+fn read_vehicle_journeys(
+    collections: &mut Collections,
+    context: &mut NetexContext,
+    vehicle_journeys: &Element,
+) {
+    for node in vehicle_journeys.children() {
+        match node.name() {
+            "ServiceJourney" => read_service_journey(collections, context, node),
+            _ => println!("Reading {} not implemented", node.name()),
```

Comment:
I intended to do so, but I could manage to see the warn when launching the `cargo test`. Digging a little further, I found this issue: https://github.com/rust-lang/rust/issues/40298
Do you think I should use both `println!` and `warn!`until the issue is resolved ?

Reply:
Discussing this point with @TeXitoi , we agreed it would be better to have only a `warn!` and keeping trace of remaining work in an issue or a deactivated test.